### PR TITLE
feat: add rule `no-dupe-keys` (fixes #108)

### LINKED ISSUE #108: Add rule to disallow duplicate type keys
Similar to http://eslint.org/docs/rules/no-dupe-keys

Bad
```javascript
type PropsType = {  
  data: Object,  
  currentValue: string,  
  data: Object,  
  onChange: Function,  
};
```
Good
```javascript
type PropsType = {
  data: Object,
  currentValue: string,
  onChange: Function,
};
```

### PR DESCRIPTION
**Pull request by aaron-harvey**

### CODE REVIEW

**Comment by gajus** [comment]
@aaron-harvey please rebase onto the master branch and re-submit the PR.

File: src/rules/noDupe.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import forEach from 'lodash/forEach';
```
Comment (gajus) [inline src/rules/noDupe.js]: Please import the entire lodash bundle.

We are using `babel-plugin-lodash` to rewire dependencies in a build step.

**Comment by danharper** [comment]
🎉  thanks for this :)

However currently it'll fail to report `type FooType = { a: number, a: string }`, as it's comparing the property source `a: number` with `a: string`. Could you check just the key instead?

For that it looks like you'd run into this `babel-eslint` bug too: https://github.com/babel/babel-eslint/issues/307 as there's currently no way to access the key of the property directly from the AST 😞 - you could probably parse out the raw tokens instead e.g. with `sourceCode.getFirstToken(node)` etc.

**Comment by aaron-harvey** [comment]
I've applied the lodash change, changed from checking the line to the property, and added a test to cover `type FooType = { a: number, a: string }` case. :tada: 

Also, sorry for any alert spam.. I messed up the rebase :disappointed:

**Comment by danharper** [comment]
This looks good, thanks :) 👍 Would you mind renaming it to `no-dupe-keys` to match ESLint's rule?

**Comment by gajus** [comment]
Thank you @aaron-harvey

**Comment by danharper** [comment]
Released as 2.18.0 - thanks! :)

**Comment by gajus** [comment]
This continuous release is literally saving me hours every week. :-)

**Comment by danharper** [comment]
Ha yeah, it's pretty awesome :)

**Pull request merged**